beater/onboarding.go

```diff
@@ -26,16 +27,28 @@ import (
 
 	"github.com/elastic/apm-server/beater/config"
 	logs "github.com/elastic/apm-server/log"
+	"github.com/elastic/apm-server/publish"
+	"github.com/elastic/apm-server/transform"
 )
 
-func notifyListening(config *config.Config, pubFct func(beat.Event)) {
+func notifyListening(ctx context.Context, config *config.Config, reporter publish.Reporter) {
 	logp.NewLogger(logs.Onboarding).Info("Publishing onboarding document")
-	event := beat.Event{
-		Timestamp: time.Now(),
+	reporter(ctx, publish.PendingReq{
```

Comment:
why not pass `context.Background()` directly to the `reporter` call instead of making it an argument of `notifyListening`?

Reply:
`notifyListening` is no longer called in the background (i.e. in its own goroutine). Passing it in from the caller means `notifyListening` doesn't make any assumptions about its calling context.

It doesn't matter much here as it's unlikely we'd ever change this code substantially, but in general it's best to stick to the pattern of instantiating a background context at the top of the stack, passing it down. Then you can add spans to a trace, for example.